### DISCUSSION TITLE
Questions about model selection

### DISCUSSION THREAD

**Discussion by emmats** [atsa-es/fish550-2023#13] [General]
I have so many questions but I will miss class tomorrow as I am traveling with my family.
1. I'm trying to understand how to determine if auto.arima is really choosing an appropriate model. For instance: If my ACF trails off and my PACF has a significant lag at 6, and my data requires d=1 I would guess I p = some kind of big number (5?) and q=0. auto.arima suggests 4,1,1. My forecasted data is not awesome, so should I try to force a model closer to what ACF and PACF indicate?
2. In comparing models with stepwise and approximation turned on or off I sometimes get very different suggestions. Is there value in going through the step with stepwise and approximation = T?

I have more questions but can save them for Thursday. Thank you!

**Comment by eeholmes** [comment]
Hi Emma, 

Definitely I would not assume that auto.arima() is selecting the model with the best forecast performance. It **should** be picking the best model vis-a-vis whatever model selection criteria you choose (default is AICc) as long as you turn off stepwise and approximation. But best in terms of AICc (or AIC or BIC) does not necessarily mean that the forecast performance is the best. How the AICc order and the forecast performance compare is an interesting question. 

You can fit specfic ARIMA models with the Arima() function and then compare forecast performance for the models.

```
library(forecast)
train <- window(WWWusage, start=1, end=90)
test <- window(WWWusage, start=91, end=100)
fit1 <- auto.arima(train, trace=TRUE)
```
ARIMA(2,0,2) with non-zero mean

Fit with ARIMA
```
fit <- Arima(WWWusage, order=c(2,0,2), include.mean=TRUE)
```

Now you can get the accuracy for the forecast
```
accuracy(forecast(fit, h=10), test)
```

Now you can compare the test accuracy (2nd line in output above) for the different models in returned in trace
```
ARIMA(2,0,2) with non-zero mean : 473.0118
 ARIMA(0,0,0) with non-zero mean : 881.4329
 ARIMA(1,0,0) with non-zero mean : 579.9167
 ARIMA(0,0,1) with non-zero mean : Inf
 ARIMA(0,0,0) with zero mean     : 1136.751
 ARIMA(1,0,2) with non-zero mean : 492.9468
 ARIMA(2,0,1) with non-zero mean : 474.2132
 ARIMA(3,0,2) with non-zero mean : Inf
 ARIMA(2,0,3) with non-zero mean : Inf
 ARIMA(1,0,1) with non-zero mean : 509.2723
 ARIMA(1,0,3) with non-zero mean : 483.8352
 ARIMA(3,0,1) with non-zero mean : 476.0242
 ARIMA(3,0,3) with non-zero mean : Inf
 ARIMA(2,0,2) with zero mean     : Inf
```
and compare the AICc order to the forecast performance order.

BTW the "Inf" values are not real Infs. You can fit the model with `Arima()` and get the real AICc.
